### DISCUSSION TITLE
HOW TO Amend files in older commit

### DISCUSSION THREAD

**Discussion by pete4abw** [community/community#21623] [New to GitHub]
<p>I have a local branch, not merged or pushed to github. I made successive changes to the branch with non-overlapping commit items.</p>
<p>Branch-&gt;HEAD: File 3<br>
Branch-&gt;HEAD~1: File 2<br>
Branch-&gt;HEAD~2: File 1</p>
<p>I found an issue in File 1 and edited it. Can I somehow use <code>git commit --amend</code> to change HEAD~2 without upsetting the two later commits? I don’t want to have a new small HEAD commit. Some version of <code>git checkout</code> or <code>git rebase</code> perhaps?</p>
<p>Since this is only a local branch, not pushed, I don’t care about the commit history being changed.</p>
<p>Thanks.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><code>git rebase -i HEAD~3</code> is probably what you’re looking for. The base commit for the rebase must be the parent of the first one you want to edit. The option <code>-i</code> (short for <code>--interactive</code>) lets you choose what to do to each commit since, with <code>edit</code> being one option.</p>
<p>See <a href="https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History" class="inline-onebox">Git - Rewriting History</a> for a lot more details. :slightly_smiling_face:</p>

**Comment by pete4abw** [comment]
<p><strong>EDIT</strong>: Just read your link. Looks like that will do the trick. Thank you. Will post update.</p>
<p>After that, how do I return to the current HEAD? Are the following changes preserved? Rebase had always made me nervous :slight_smile:</p>
<p>Thanks</p>

**Comment by pete4abw** [comment]
<p>I committed the change to be merged with HEAD~2. Did the <code>git rebase -i HEAD~3</code> command and reordered and squashed the two commits. It looks like it worked. Did a backup of the local repo anyway just in case. Thanks.</p>
